src/flint/fmpz.jl

```diff
@@ -942,16 +942,23 @@ end
 ###############################################################################
 
 @doc Markdown.doc"""
-    gcd(x::fmpz, y::fmpz)
+    gcd(x::fmpz, y::fmpz, z::fmpz...)
 
-Return the greatest common divisor of $x$ and $y$. The returned result will
-always be nonnegative and will be zero iff $x$ and $y$ are zero.
+Return the greatest common divisor of $(x, y, ...)$. The returned result will
+always be nonnegative and will be zero iff all inputs are zero.
 """
-function gcd(x::fmpz, y::fmpz)
-   z = fmpz()
+function gcd(x::fmpz, y::fmpz, z::fmpz...)
+   d = fmpz()
    ccall((:fmpz_gcd, libflint), Nothing,
-         (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}), z, x, y)
-   return z
+         (Ref{fmpz}, Ref{fmpz}, Ref{fmpz}), d, x, y)
+   length(z) == 0 && return d
+
+   for ix in 1:length(z)
+     tmp = d
```

Comment:
Is this line necessary? Can't you just take the gcd of d and z[ix] below?

Reply:
Sorry, of course. Thank you!